checker missed the check on the initialization of the result field of the struct

V doctor:

```
OS: macos, macOS, 12.6, 21G115
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8569U CPU @ 2.80GHz
CC version: Apple clang version 14.0.0 (clang-1400.0.29.102)

getwd: /usr/local/v
vmodules: /Users/shove/.vmodules
vroot: /usr/local/v
vexe: /usr/local/v/v
vexe mtime: 2022-10-22 15:26:59
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.1 e863191.5913a93

Git version: git version 2.37.0 (Apple Git-136)
Git vroot status: weekly.2022.41-48-g5913a93a-dirty
.git/config present: true
thirdparty/tcc status: thirdparty-macos-amd64 46662e20
```

**What did you do?**

```v
struct Foo {
	bar !int
}

fn main() {
	_ := Foo {
		bar: 1
	}
}
```

**What did you expect to see?**

```
error:  field `bar` is result, but initialization of result fields currently unsupported
 ```

**What did you see instead?**

```
==================
/tmp/v_501/struct_field_is_optional_err.7711547511520011829.tmp.c:1076:14: error: field has incomplete type '_result_int' (aka 'struct _result_int')
        _result_int bar;
                    ^
/tmp/v_501/struct_field_is_optional_err.7711547511520011829.tmp.c:71:16: note: forward declaration of 'struct _result_int'
typedef struct _result_int _result_int;
               ^
/tmp/v_501/struct_field_is_optional_err.7711547511520011829.tmp.c:2153:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
1 warning and 1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/checker/struct.v
+++ b/vlib/v/checker/struct.v
@@ -87,7 +87,7 @@ pub fn (mut c Checker) struct_decl(mut node ast.StructDecl) {
 			if field.has_default_expr {
 				c.expected_type = field.typ
 				default_expr_type := c.expr(field.default_expr)
-				if !field.typ.has_flag(.optional) {
+				if !field.typ.has_flag(.optional) && !field.typ.has_flag(.result) {
 					c.check_expr_opt_call(field.default_expr, default_expr_type)
 				}
 				struct_sym.info.fields[i].default_expr_typ = default_expr_type
@@ -420,7 +420,7 @@ pub fn (mut c Checker) struct_init(mut node ast.StructInit) ast.Type {
 				if expr_type == ast.void_type {
 					c.error('`$field.expr` (no value) used as value', field.pos)
 				}
-				if !field_info.typ.has_flag(.optional) {
+				if !field_info.typ.has_flag(.optional) && !field.typ.has_flag(.result) {
 					expr_type = c.check_expr_opt_call(field.expr, expr_type)
 				}
 				expr_type_sym := c.table.sym(expr_type)
@@ -463,6 +463,10 @@ pub fn (mut c Checker) struct_init(mut node ast.StructInit) ast.Type {
 
 				if field_info.typ.has_flag(.optional) {
 					c.error('field `$field_info.name` is optional, but initialization of optional fields currently unsupported',
+						field.pos)
+				}
+				if field_info.typ.has_flag(.result) {
+					c.error('field `$field_info.name` is result, but initialization of result fields currently unsupported',
 						field.pos)
 				}
 				if expr_type.is_ptr() && expected_type.is_ptr() {

--- a/vlib/v/checker/tests/struct_field_optional_init_err.out
+++ b/vlib/v/checker/tests/struct_field_optional_init_err.out
@@ -1,0 +1,14 @@
+vlib/v/checker/tests/struct_field_optional_init_err.vv:8:3: error: field `bar` is result, but initialization of result fields currently unsupported
+    6 | fn main() {
+    7 |     _ := Foo {
+    8 |         bar: 1
+      |         ~~~~~~
+    9 |         baz: 1
+   10 |     }
+vlib/v/checker/tests/struct_field_optional_init_err.vv:9:3: error: field `baz` is optional, but initialization of optional fields currently unsupported
+    7 |     _ := Foo {
+    8 |         bar: 1
+    9 |         baz: 1
+      |         ~~~~~~
+   10 |     }
+   11 | }

--- a/vlib/v/checker/tests/struct_field_optional_init_err.out
+++ b/vlib/v/checker/tests/struct_field_optional_init_err.out
@@ -1,12 +1,12 @@
 vlib/v/checker/tests/struct_field_optional_init_err.vv:8:3: error: field `bar` is result, but initialization of result fields currently unsupported
     6 | fn main() {
-    7 |     _ := Foo {
+    7 |     _ := Foo{
     8 |         bar: 1
       |         ~~~~~~
     9 |         baz: 1
    10 |     }
 vlib/v/checker/tests/struct_field_optional_init_err.vv:9:3: error: field `baz` is optional, but initialization of optional fields currently unsupported
-    7 |     _ := Foo {
+    7 |     _ := Foo{
     8 |         bar: 1
     9 |         baz: 1
       |         ~~~~~~

--- a/vlib/v/checker/tests/struct_field_optional_init_err.vv
+++ b/vlib/v/checker/tests/struct_field_optional_init_err.vv
@@ -1,0 +1,11 @@
+struct Foo {
+	bar !int
+	baz ?int
+}
+
+fn main() {
+	_ := Foo{
+		bar: 1
+		baz: 1
+	}
+}

